firecracker-control/local.go

```diff
@@ -207,10 +235,35 @@ func (s *local) buildVMConfiguration(ctx context.Context, id string, cid uint32,
 		driveBuilder = drivesBuilderFromProto(drive, driveBuilder, false)
 	}
 
-	// TODO: Reserve fake drives here (https://github.com/firecracker-microvm/firecracker-containerd/pull/154)
+	s.stubDriveIndex = len(driveBuilder.Build()) - 1
+	tempDir, err := ioutil.TempDir("", "stub-drives")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create temp directory")
+	}
+
+	instance, err := s.getVM(id)
+	if err != nil {
+		return nil, err
+	}
 
-	cfg.Drives = driveBuilder.Build()
+	instance.cleanupFns = append(instance.cleanupFns, func() error {
+		return os.RemoveAll(tempDir)
+	})
+
+	paths, err := s.stubDriveHandler.StubDrivePaths(int(req.ContainerCount))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to retrieve stub drive paths")
+	}
+
+	for i, path := range paths {
+		driveID := fmt.Sprintf("%s%d", internal.MagicStubBytes, i)
+		driveBuilder = driveBuilder.AddDrive(path, false, func(drive *models.Drive) {
+			drive.DriveID = firecracker.String(driveID)
+		})
+	}
 
+	cfg.Drives = driveBuilder.Build()
+	s.drives = cfg.Drives
```

Comment:
How this logic is going to work? Each new VM will overwrite `s.drives`, so we no longer able to patch drives of previous VM?